src/drtsans/tof/eqsans/correct_frame.py

```diff
@@ -649,13 +656,29 @@ def transform_to_wavelength(
     if output_workspace is None:
         output_workspace = str(input_workspace)
 
-    # generate bands if not given
+    # generate bands from input workspace
+    if low_tof_clip > 0.0 or high_tof_clip > 0.0:
+        sdd = source_detector_distance(input_workspace, unit="m")
+        bands_from_ws = transmitted_bands_clipped(input_workspace, sdd, low_tof_clip, high_tof_clip)
+    else:
+        bands_from_ws = transmitted_bands(input_workspace)
+
+    # use generated bands if not given
     if bands is None:
-        if low_tof_clip > 0.0 or high_tof_clip > 0.0:
-            sdd = source_detector_distance(input_workspace, unit="m")
-            bands = transmitted_bands_clipped(input_workspace, sdd, low_tof_clip, high_tof_clip)
-        else:
-            bands = transmitted_bands(input_workspace)
+        bands = bands_from_ws
+    else:  # check that the given bands are compatible with the bands found from the workspace
+        diff_bands_min = np.abs(bands.lead.min - bands_from_ws.lead.min)
```

Comment:
can you add method `almost_equal(self, other, atol=1e-6)` for classes `WBands` and `WBand`? 
After this, you can refactor as:
```python
if bands.almost_equal(bands_from_ws, atol=WAVELENGTH_BAND_DIFF_TOLERANCE) is False:
    raise IncompatibleWavelengthBandsError(f"bands differ by more than {WAVELENGTH_BAND_DIFF_TOLERANCE} Angstroms")
```

Reply:
good idea!